executor/diagnostics_test.go

```diff
@@ -166,6 +171,140 @@ func (s *diagnosticsSuite) TestInspectionResult(c *C) {
 	}
 }
 
+func (s *diagnosticsSuite) TestThresholdCheckInspection(c *C) {
+	// mock TiKV http server
+	router := mux.NewRouter()
+	type mockServer struct {
+		address string
+		server  *httptest.Server
+	}
+	server := httptest.NewServer(router)
+	address := strings.TrimPrefix(server.URL, "http://")
+	testServer := &mockServer{
+		address: address,
+		server:  server,
+	}
+	defer func() { testServer.server.Close() }()
+	var mockTiKVConfig = func() (map[string]interface{}, error) {
+		configuration := map[string]interface{}{
+			"raftstore.apply-pool-size":               "2",
+			"raftstore.store-pool-size":               "2",
+			"readpool.coprocessor.normal-concurrency": "4",
+			"readpool.coprocessor.high-concurrency":   "4",
+			"readpool.coprocessor.low-concurrency":    "4",
+			"readpool.storage.high-concurrency":       "4",
+			"readpool.storage.low-concurrency":        "4",
+			"readpool.storage.normal-concurrency":     "4",
+			"server.grpc-concurrency":                 "8",
+			"storage.scheduler-worker-pool-size":      "6",
+		}
+		return configuration, nil
+	}
+	router.Handle("/config", fn.Wrap(mockTiKVConfig))
```

Comment:
We should use `inspection_schema.cluster_config` in inspect mode (and it's easier to mock data in `inspection_schema.cluster_config`).